### DISCUSSION TITLE
TIC-80 tiny computer 1.0.2164

### DISCUSSION THREAD

**Discussion by nesbox** [nesbox/TIC-80#1894] [General]
## What's Changed
* Added --version command line argument by @nalquas in https://github.com/nesbox/TIC-80/pull/1503
* Handle emacs ^n ^p and ^e keys in the editor by @trufae in https://github.com/nesbox/TIC-80/pull/1530
* sdl: Fix comparision warning by @RobLoach in https://github.com/nesbox/TIC-80/pull/1537
* Add TIC_UNUSED() to handle unused parameter warnings by @RobLoach in https://github.com/nesbox/TIC-80/pull/1538
* Update Fennel to version 0.10.0. by @technomancy in https://github.com/nesbox/TIC-80/pull/1539
* fix path problems with baremeta #1541 by @msx80 in https://github.com/nesbox/TIC-80/pull/1582
* Improvements for language support by @msx80 in https://github.com/nesbox/TIC-80/pull/1590
* Baremetal RPI: added Squirrel language which was previously disabled. by @msx80 in https://github.com/nesbox/TIC-80/pull/1596
* Fix eval command by @technomancy in https://github.com/nesbox/TIC-80/pull/1633
* Optimize tic api call performance in LUA. by @vsariola in https://github.com/nesbox/TIC-80/pull/1628
* #714: fix crackling audio. by @vsariola in https://github.com/nesbox/TIC-80/pull/1618
* Activate strict mode in Fennel when strict: true header is found. by @technomancy in https://github.com/nesbox/TIC-80/pull/1653
* Poke1 and Peek1 by @SuperPupperDoggo in https://github.com/nesbox/TIC-80/pull/1616
* (fix) ignore negative index passed to spr by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1665
* (enh) errors when mkdir fails by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1667
* (chore) move moonscript into separate C file by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1670
* fixup: move moon script and fennel into sep files by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1671
* (enh) Music editor: Skip over sharp column when using cursor keys by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1672
* Fedora build instructions by @lzap in https://github.com/nesbox/TIC-80/pull/1679
* add MacOS application instructions by @lzap in https://github.com/nesbox/TIC-80/pull/1680
* suggest pro when trying to open source with standard by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1684
* Replace a right single apostrophe in the API help with an apostrophe by @blinry in https://github.com/nesbox/TIC-80/pull/1697
* Improvements to README.md by @Tyler887 in https://github.com/nesbox/TIC-80/pull/1701
* Add community guidelines by @Tyler887 in https://github.com/nesbox/TIC-80/pull/1703
* Update Fennel to version 1.0.0. by @technomancy in https://github.com/nesbox/TIC-80/pull/1708
* squirrel: Fix mouse() from returning null values by @RobLoach in https://github.com/nesbox/TIC-80/pull/1719
* Add some missing spaces, and reword a few phrases by @blinry in https://github.com/nesbox/TIC-80/pull/1722
* sdl: Fix warnings of string.h missing by @RobLoach in https://github.com/nesbox/TIC-80/pull/1724
* Fix OpenDingux builds by @RobLoach in https://github.com/nesbox/TIC-80/pull/1725
* Ruby scripting support by @remi6397 in https://github.com/nesbox/TIC-80/pull/1726
* Small changes to the Ruby bindings by @remi6397 in https://github.com/nesbox/TIC-80/pull/1733
* Add Shift+Return shortcut to the code editor that auto-closes a block by @blinry in https://github.com/nesbox/TIC-80/pull/1721
* When listing language options in the spec, use "x, y, or z" format by @blinry in https://github.com/nesbox/TIC-80/pull/1745
* On Linux, use Ctrl+L to clear the console, per default of other shells by @blinry in https://github.com/nesbox/TIC-80/pull/1753
* Add help topics for keys and buttons by @blinry in https://github.com/nesbox/TIC-80/pull/1754
* Typo: Correct variable was not getting updated by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1783
* Create SECURITY.md by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1786
* Console: `cd` with no args returns home by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1789
* add default license, version, site tags to demos by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1790
* support CTRL-A, CTRL-E console editing keystrokes by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1791
* (chore) bring all Demo header into sync with new tags by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1793
* fix: prevent cartridges hijacking keyboard events by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1787
* Do not default to the first available language for new carts. by @remi6397 in https://github.com/nesbox/TIC-80/pull/1792
* (enh) moonscript supports eval now by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1800
* faster drawTile, remove repeat clipping checks by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1801
* order langauges by alphabetical order by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1804
* split out main menu code from studio by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1813
* Fix compilation for emscripten libretro by @phcoder in https://github.com/nesbox/TIC-80/pull/1821
* mruby: Make Ruby Build Optional by @RobLoach in https://github.com/nesbox/TIC-80/pull/1823
* Update SDL2 version by @luizdepra in https://github.com/nesbox/TIC-80/pull/1825
* Export without editor in pro version is done. by @Darenn in https://github.com/nesbox/TIC-80/pull/1837
* (chore) update Zig source to use new BOOT() by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1863
* [WIP] Advanced tab-completion for the console by @blinry in https://github.com/nesbox/TIC-80/pull/1840
* fix(zig) fix template compiling and add TextureSource by @joshgoebel in https://github.com/nesbox/TIC-80/pull/1850
* D WebAssembly binding by @PierceNg in https://github.com/nesbox/TIC-80/pull/1886
* Fix crash when internet is disabled by @OldiLo in https://github.com/nesbox/TIC-80/pull/1890
* Improved D template by @PierceNg in https://github.com/nesbox/TIC-80/pull/1891

## New Contributors
* @trufae made their first contribution in https://github.com/nesbox/TIC-80/pull/1530
* @vsariola made their first contribution in https://github.com/nesbox/TIC-80/pull/1628
* @lzap made their first contribution in https://github.com/nesbox/TIC-80/pull/1679
* @blinry made their first contribution in https://github.com/nesbox/TIC-80/pull/1697
* @Tyler887 made their first contribution in https://github.com/nesbox/TIC-80/pull/1701
* @remi6397 made their first contribution in https://github.com/nesbox/TIC-80/pull/1726
* @luizdepra made their first contribution in https://github.com/nesbox/TIC-80/pull/1825
* @Darenn made their first contribution in https://github.com/nesbox/TIC-80/pull/1837
* @PierceNg made their first contribution in https://github.com/nesbox/TIC-80/pull/1886
* @OldiLo made their first contribution in https://github.com/nesbox/TIC-80/pull/1890

**Full Changelog**: https://github.com/nesbox/TIC-80/compare/0.90.1723...v1.0.2164

<hr /><em>This discussion was created from the release <a href='https://github.com/nesbox/TIC-80/releases/tag/v1.0.2164'>TIC-80 tiny computer 1.0.2164</a>.</em>

**Comment by blinry** [comment]
Yay, congratulations on the 1.0 release @nesbox and everyone! :tada:

**Comment by rayxsignal** [comment]
Hi! I have installed the current version on all my devices, including 3DS. I have a question about the new logic of the Esc key. When I press Esc while the application is running, I don't get into the console or the code editor, but to a separate window for selecting actions. Also, the function of a quick transition from a running application in the editor by pressing the keys F2, F3, F4 has stopped working. Tell me, is it possible to return the old behavior of the buttons?
The new behavior greatly slows down the work with the program.

**Reply by blinry** [reply]
I agree, this is suboptimal. In the Options, you can turn the "dev mode" on, @rayxsignal, to restore the old behavior.

But this also really confused the people in a TIC-80 workshop I gave recently. Should we open an issue suggesting that the "new behavior" should be the default for exported games, and the "old behavior" should be the default in all other circumstances?

**Reply by nesbox** [reply]
Let's leave dev mode as is, because people also play games from the Surf where dev mode should be disabled by default.

**Reply by rayxsignal** [reply]
I support this decision. I just didn't know how to change the behavior. Now I figured it out, and personally everything suits me.

**Reply by blinry** [reply]
I'd still strongly argue that dev mode should be on when booting into the console, to make development smoother for everyone, while preserving the benefit of giving access to the options in the context of a deployed game (or in surf).

I tried to describe the problem and the proposed solution in #1950, let me know what you think! :)

**Comment by btaratuta** [comment]
Which one of these files works with a Pi3

**Reply by jamesgecko** [reply]
One of the `rpi` files. You probably want the `.deb` download if you're running Raspberry Pi OS.
